ProcessesApi/V1/Services/SoleToJointService.cs

```diff
@@ -208,13 +233,22 @@ protected override void SetUpStates()
                     .Permit(SoleToJointPermittedTriggers.RequestDocumentsAppointment, SoleToJointStates.DocumentsRequestedAppointment);
 
             _machine.Configure(SoleToJointStates.DocumentsRequestedDes)
-                    .Permit(SoleToJointPermittedTriggers.RequestDocumentsAppointment, SoleToJointStates.DocumentsRequestedAppointment);
+                    .InternalTransitionAsync(SoleToJointPermittedTriggers.ReviewDocuments, async (x) => await ReviewDocumentsCheck(x).ConfigureAwait(false))
+                    .Permit(SoleToJointInternalTriggers.DocumentChecksPassed, SoleToJointStates.DocumentChecksPassed)
+                    .Permit(SoleToJointPermittedTriggers.RequestDocumentsAppointment, SoleToJointStates.DocumentsRequestedAppointment)
+                    .Permit(SoleToJointPermittedTriggers.CloseProcess, SharedProcessStates.ProcessClosed);
 
             _machine.Configure(SoleToJointStates.DocumentsRequestedAppointment)
-                    .Permit(SoleToJointPermittedTriggers.RescheduleDocumentsAppointment, SoleToJointStates.DocumentsAppointmentRescheduled);
+                    .InternalTransitionAsync(SoleToJointPermittedTriggers.ReviewDocuments, async (x) => await ReviewDocumentsCheck(x).ConfigureAwait(false))
```

Comment:
Turns out you don't need to have the whole lambda expression, you can just have something like this:

```
.InternalTransitionAsync(SoleToJointPermittedTriggers.ReviewDocuments, ReviewDocumentsCheck)
```